src/Sparrow.Server/Compression/Encoder3Gram.cs

```diff
@@ -363,7 +368,12 @@ private void BuildDictionary(in FastList<SymbolCode> symbolCodeList)
             // We haven't stored it yet. So we are calculating it. 
             int numberOfEntriesInTable = (_state.EncodingTable.Length - 4) / Unsafe.SizeOf<Interval3Gram>();
             if (numberOfEntriesInTable < dictSize)
-                throw new ArgumentException("Not enough memory to store the dictionary");
+            {
+                if (_state.CanGrow)
+                    _state.Grow(dictSize);
+                else
+                    throw new ArgumentException("Not enough memory to store the dictionary");
```

Comment:
Add how much memory is required? And what is the current size?

Reply:
1. I believe Oren meant to adding some extra info to the exception message so we can have more details when it happens again.
2. `ArgumentException` doesn't seem to be suitable here.